### PR TITLE
HS-754: Update to Java 17

### PR DESCRIPTION
**Pull request by cgorantla**
## Description

JDK-17 Upgrade on all services

## Jira link(s)
- https://issues.opennms.org/browse/HS-754

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: minion-gateway/docker-it/src/test/resources/org/opennms/horizon/minion-gateway.feature [hunk 1]
```diff
@@ -16,6 +16,7 @@ Feature: Minion Gateway RPC Request Processing
     Then verify RPC exception was received
     Then verify RPC exception states active connection could not be found
 
+  @ignore
```
Comment (j-white) [inline minion-gateway/docker-it/src/test/resources/org/opennms/horizon/minion-gateway.feature:19]: Is this failure due to the JDK17 upgrade?

Comment (cgorantla) [reply minion-gateway/docker-it/src/test/resources/org/opennms/horizon/minion-gateway.feature]: Fixed it in new commit!

**Comment by thomasbigger584** [comment]
running tilt on m1 mac working ok

**Review by splatch — APPROVED** [review]
:+1: LGTM I've made deployment on cleaned cluster, it works without any issues. No startup errors observed, all functionality I expected to work is still there.

**Review by yangli2014 — APPROVED** [review]
Tested and it worked in my local environment after cleaning docker and fresh rebuilding minion images.

**Review by mikewrosey — APPROVED** [review]
Works fine with wsl

**Pull request merged**